presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/DropStatsProcedure.java

```diff
@@ -87,20 +89,23 @@ public Procedure get()
                 DROP_STATS.bindTo(this));
     }
 
-    public void dropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    public void dropStats(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<?> partitionValues)
     {
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
-            doDropStats(session, schema, table, partitionValues);
+            doDropStats(session, accessControl, schema, table, partitionValues);
         }
     }
 
-    private void doDropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    private void doDropStats(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<?> partitionValues)
     {
         TransactionalMetadata hiveMetadata = hiveMetadataFactory.create();
         HiveTableHandle handle = (HiveTableHandle) hiveMetadata.getTableHandle(session, new SchemaTableName(schema, table));
         if (handle == null) {
             throw new PrestoException(INVALID_PROCEDURE_ARGUMENT, format("Table '%s' does not exist", new SchemaTableName(schema, table)));
         }
+
+        accessControl.checkCanDeleteFromTable(null, new SchemaTableName(schema, table));
```

Comment:
Let's make this `checkCanInsertIntoTable` to match `ANALYZE`, since we're not deleting data. This would mean a user who has insert permission can drop stats and then run analyze.

From the connector's point of view, analyze actually overwrites (i.e. deletes) the existing statistics, so drop stats can be thought of as a limited form of analyze.

Separately, we should probably add a new permission for analyze.